beamer/state_machine.py

```diff
@@ -367,9 +367,17 @@ def _handle_initiate_l1_resolution(
 
     # A request should never be dropped before all claims are finalized
     assert request is not None, "Request object missing"
-
     assert request.fill_tx is not None, "Request not yet filled"
-    if _l1_resolution_criteria_fulfilled(claim, context):
+    assert request.fill_timestamp is not None, "Request not yet filled"
+
+    # Check that message is finalized on L1
+    if not _timestamp_is_l1_finalized(
```

Comment:
this will cost one RPC call per second for each claim whose request was filled (can also be handled in a later PR)

Reply:
Created #784 for this